checkov/terraform/checks/resource/azure/AKSEncryptionAtHostEnabled.py

```diff
@@ -22,14 +22,23 @@ def __init__(self) -> None:
             id=id,
             categories=categories,
             supported_resources=supported_resources,
-            missing_block_result=CheckResult.FAILED,
         )
 
-    def get_inspected_key(self) -> str:
+    def scan_resource_conf(self, conf):
         if self.entity_type == "azurerm_kubernetes_cluster":
-            return "default_node_pool/[0]/enable_host_encryption"
+            if conf.get('default_node_pool'):
+                node_pool = conf['default_node_pool'][0]
+                if (node_pool.get('enable_host_encryption') == [True] or
+                        node_pool.get('host_encryption_enabled') == [True]):
+                    return CheckResult.PASSED
+            self.evaluated_keys = ['default_node_pool/[0]/enable_host_encryption',
```

Comment:
Do we need to update the evaluated_keys here? What is the reason?

Reply:
For highlighting in the platform (and fix suggestions, but not relevant for this check)